packages/app/app/actions/favorites.ts

```diff
@@ -119,3 +119,48 @@ export function removeFavoriteArtist(artist) {
     payload: favorites
   };
 }
+
+/**
+* Helper function to read the old track format into the new format.
+*
+* `Track.artist` and `Track.extraArtists` are written into {@link Track.artists}
+*/
+function getFavoritesBackwardsCompatible() {
+  const favorites = store.get('favorites');
+
+  favorites.tracks?.forEach(track => {
+    if (track.artists || !track.artist) {
+      return;
+    }
+
+    if (track.artist) {
+      track.artists = _.isString(track.artist) ? [track.artist] : [track.artist.name];
+    }
+
+    // Assuming we have `extraArtists` on a track, we must had an `artist` which
+    // was already saved into `artists`, so this `track.artists` shouldn't be undefined
+    track.extraArtists?.forEach(artist => {
+      track.artists.push(artist);
+    });
+  });
+
+  favorites.albums?.forEach(album => {
+    album.tracklist?.forEach(track => {
+      if (track.artists || !track.artist) {
+        return;
+      }
+
+      if (track.artist) {
+        track.artists = _.isString(track.artist) ? [track.artist] : [track.artist.name];
+      }
+
+      // Assuming we have `extraArtists` on a track, we must had an `artist` which
+      // was already saved into `artists`, so this `track.artists` shouldn't be undefined
+      track.extraArtists?.forEach(artist => {
+        track.artists.push(artist);
+      });
+    });
+  });
+
```

Comment:
This part can use the function that can be extracted from all `getXBackwardsCompatible`.

Reply:
As mentioned in the Discord thread, this is very strange.
I moved it in a separate helper function and then the tests start failing because the track is suddenly undefined.